### PR TITLE
User/jf/add cached ci yml

### PR DESCRIPTION
**Pull request by feng-j678**
- add cached for py27 installation
- add cached for install coverage

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -101,4 +119,5 @@ jobs:
         if: contains( steps.getoutput.outputs.contents, 'FAILED (failures=' )
         run: |
           echo "${{ steps.getoutput.outputs.contents }}"
+          echo "There are failed tests"
```
Comment (kjohn-msft) [inline .github/workflows/ci.yml:122]: Why has this changed?

**Review by kjohn-msft — COMMENTED** [review]
Looks good - one open question

File: .github/workflows/ci.yml (hunk 1 above)
Comment (feng-j678) [reply .github/workflows/ci.yml:122]: this message is just to let user know there are some failed tests. I can remove it if you think it's unnecessary.

Comment (kjohn-msft) [reply .github/workflows/ci.yml:122]: Sorry, when I'd left the comment, it looked like line 121 was replaced (not augmented) with 122. Nvm.

**Review by kjohn-msft — APPROVED** [review]

**Review by rane-rajasi — APPROVED** [review]

**Pull request merged**